api.py

```diff
@@ -54,3 +54,80 @@ def create_poem():
 
     result = {'created': True, 'poem': poem}
     return json.dumps(result, cls=models.ModelEncoder)
+
+
+@blueprint.route('/get_poems/<any(best, new):poem_type>/<int:num_poems>',
+                 defaults={'user_email': None})
+@blueprint.route(
+    '/get_poems/<any(manual, generated):poem_type>/<int:num_poems>/<user_email>'
+)
+def get_poems(poem_type, num_poems, user_email):
+    """Retreives poems
+
+    Args:
+        poem_type (string): one of best, new, manual, or generated
+            best: returns poem with the highest likes - dislikes
+            new: returns poems with the most recent creation timestamp
+            manual: returns poems that the user manually wrote
+            generated: returns poems that the user generated
+        num_poems (int): the number of poems to retrieve. 0 returns all poems
+        user_email (string): The email of the user to retrieve poems for.
```

Comment:
Is there value in allowing retrieving the best and new poems for a particular user?

Reply:
I think if the My Poems page wants to support sorting the written/generated poems it could be useful, but that may need to be implemented as a sort order parameter of the `manual` or `generated` poem types. Right now, the `new` and `best` types are for the home page so there's no user filtering.

Reply:
After looking at the future tasks, I added another case for the user's liked poems.

I think that's still a separate case from the 'best' poems since they SQL queries will be very different.